frontend/src/components/auth/utils/validator.jsx

```diff
@@ -0,0 +1,47 @@
+import { PASSWORD_REGEX, EMAIL_REGEX } from "../../../constants";
+import { RECAPTCHA_SITEKEY } from "../../../constants/environment";
+
+export function PasswordValidator(password, confirmPassword) {
+  const errors = {};
+  // password fields
+  if (!password) {
+    errors.password = "Required";
+  } else if (password.length < 12) {
+    errors.password = "Must be 12 characters or more";
+  } else if (!PASSWORD_REGEX.test(password)) {
+    errors.password =
+      "The password is entirely numeric or contains special characters";
+  }
+  if (!confirmPassword) {
+    errors.confirmPassword = "Required";
+  } else if (confirmPassword.length < 12) {
+    errors.confirmPassword = "Must be 12 characters or more";
+  }
+  if (
+    password.length > 0 &&
+    confirmPassword.length > 0 &&
+    password !== confirmPassword
+  ) {
+    errors.password = "Passwords do not match.";
+    errors.confirmPassword = "Passwords do not match.";
+  }
+  return errors;
+}
```

Comment:
i suggest to split this in two functions:
* First function takes a string and perform the validation of the pwd
* Second function takes two strings (the two passwords) and checks they are equal.

Then you can call the first function twice: on the password and on the confirmPassword. The second function called once with the two passwords to check they are equals.